frontend/src/metabase/modes/components/drill/QuickFilterDrill/QuickFilterDrill.tsx

```diff
@@ -77,29 +96,47 @@ const getOperatorOverrides = ({
   return null;
 };
 
-const QuickFilterDrill: Drill = ({
+const QuickFilterDrill = ({
   question,
   clicked,
-}): QuestionChangeClickAction[] => {
+}: ClickActionProps): Array<QuestionChangeClickAction | PopoverClickAction> => {
   const drill = quickFilterDrill({ question, clicked });
-  if (!drill || !clicked) {
+  if (!drill || !drill.operators || !clicked) {
     return [];
   }
 
   const { value } = clicked;
-  const { operators, valueType } = drill;
-
-  return operators.map(({ name, filter }) => ({
-    name,
-    title: name,
-    section: "filter",
-    buttonType: "token-filter",
-    question: () => quickFilterDrillQuestion({ question, clicked, filter }),
-    extra: () => ({
-      valueType,
-    }),
-    ...getOperatorOverrides({ valueType, name, value }),
-  }));
+  const { query, operators } = drill;
+
+  return operators.map(operator => {
+    const { name, filter, valueType } = operator;
+
+    if (CUSTOM_BEHAVIOR_OPERATORS.includes(name)) {
+      return {
+        name,
+        title: name,
+        section: "filter",
+        buttonType: "token-filter",
+        popover: getColumnFilterDrillPopover({
+          query,
+          initialFilter: new Filter(filter, 0, query),
```

Comment:
This line might be incorrect. It implies that we're always creating filter for the last stage of the query. Could you double check that new filter menu options work with columns from other stages? Please try queries at least with 3 levels of nesting.